### ISSUE TITLE
Make Newton work with infinite intervals

### ISSUE THREAD

**Issue by dpsanders** [JuliaIntervals/IntervalRootFinding.jl#56]
```
julia> roots(x->x^2 - 2, -Inf..Inf)
1-element Array{IntervalRootFinding.Root{IntervalArithmetic.Interval{Float64}},1}:
 Root([-∞, ∞], :unique)
``` 

This seems to be because `N(X) = X`

**Comment by dpsanders** [comment]
The problem is
```
julia> (-∞..∞) ⪽(-∞..∞)
true
```

**Comment by lbenet** [comment]
I think this is a requirement of the interval standard, but I would need to check that.

**Comment by dpsanders** [comment]
Yes, it seems to be; one of the ITF1788 tests tests exactly this.

In that case, in Newton itself we'll simply have to exclude that case.

**Comment by lbenet** [comment]
What happens if, instead of testing `X=(-∞..∞)`, you test the interval `X = interval( -prevfloat(∞), prevfloat(∞))`?

**Comment by dpsanders** [comment]
Fixed by #55.
